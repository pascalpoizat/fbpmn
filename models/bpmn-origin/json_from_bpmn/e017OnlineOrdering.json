{
    "isInterrupt": {},
    "catN": {
        "giftVoucher": "AbstractTask",
        "Process_": "Process",
        "card": "AbstractTask",
        "validatePayment": "AbstractTask",
        "email": "AbstractTask",
        "packageItems": "AbstractTask",
        "signIn": "AbstractTask",
        "initiateDelivery": "AbstractTask",
        "parallelgateway4": "AndGateway",
        "exclusivegateway1": "XorGateway",
        "generateTrackingInfo": "AbstractTask",
        "Payment": "AbstractTask",
        "inclusivegateway1": "OrGateway",
        "startevent1": "NoneStartEvent",
        "parallelgateway3": "AndGateway",
        "signUp": "AbstractTask",
        "inclusivegateway2": "OrGateway",
        "parallelgateway2": "AndGateway",
        "exclusivegateway2": "XorGateway",
        "errorendevent1": "NoneEndEvent",
        "notifyUser": "AbstractTask",
        "parallelgateway1": "AndGateway",
        "endevent1": "NoneEndEvent",
        "exclusivegateway4": "XorGateway",
        "sms": "AbstractTask",
        "payOnDelivery": "AbstractTask",
        "parallelgateway6": "AndGateway",
        "exclusivegateway3": "XorGateway",
        "transport": "AbstractTask",
        "parallelgateway5": "AndGateway",
        "selectItems": "AbstractTask",
        "initiateShipping": "AbstractTask"
    },
    "catE": {
        "flow6": "NormalSequenceFlow",
        "flow60": "NormalSequenceFlow",
        "flow14": "NormalSequenceFlow",
        "flow19": "DefaultSequenceFlow",
        "flow61": "NormalSequenceFlow",
        "flow43": "NormalSequenceFlow",
        "flow7": "NormalSequenceFlow",
        "flow57": "NormalSequenceFlow",
        "flow31": "NormalSequenceFlow",
        "flow13": "DefaultSequenceFlow",
        "flow59": "NormalSequenceFlow",
        "flow4": "DefaultSequenceFlow",
        "flow62": "NormalSequenceFlow",
        "flow30": "NormalSequenceFlow",
        "flow12": "ConditionalSequenceFlow",
        "flow56": "NormalSequenceFlow",
        "flow35": "NormalSequenceFlow",
        "flow17": "NormalSequenceFlow",
        "flow63": "NormalSequenceFlow",
        "flow41": "NormalSequenceFlow",
        "flow58": "NormalSequenceFlow",
        "flow5": "NormalSequenceFlow",
        "flow8": "NormalSequenceFlow",
        "flow33": "NormalSequenceFlow",
        "flow11": "ConditionalSequenceFlow",
        "flow16": "NormalSequenceFlow",
        "flow34": "NormalSequenceFlow",
        "flow39": "NormalSequenceFlow",
        "flow20": "ConditionalSequenceFlow",
        "flow32": "NormalSequenceFlow",
        "flow10": "NormalSequenceFlow",
        "flow9": "NormalSequenceFlow",
        "flow15": "NormalSequenceFlow",
        "flow3": "ConditionalSequenceFlow",
        "flow47": "NormalSequenceFlow",
        "flow21": "NormalSequenceFlow"
    },
    "containN": {
        "Process_": [
            "exclusivegateway4",
            "exclusivegateway3",
            "validatePayment",
            "inclusivegateway2",
            "payOnDelivery",
            "card",
            "giftVoucher",
            "inclusivegateway1",
            "Payment",
            "selectItems",
            "exclusivegateway2",
            "signUp",
            "signIn",
            "exclusivegateway1",
            "startevent1",
            "initiateShipping",
            "parallelgateway1",
            "packageItems",
            "parallelgateway2",
            "initiateDelivery",
            "email",
            "notifyUser",
            "sms",
            "transport",
            "parallelgateway5",
            "parallelgateway6",
            "parallelgateway3",
            "parallelgateway4",
            "endevent1",
            "generateTrackingInfo",
            "errorendevent1"
        ]
    },
    "messageE": {},
    "timeInformation": {},
    "targetE": {
        "flow6": "exclusivegateway2",
        "flow60": "parallelgateway6",
        "flow14": "inclusivegateway2",
        "flow19": "exclusivegateway4",
        "flow61": "parallelgateway4",
        "flow43": "transport",
        "flow7": "selectItems",
        "flow57": "email",
        "flow31": "packageItems",
        "flow13": "payOnDelivery",
        "flow59": "parallelgateway6",
        "flow4": "signUp",
        "flow62": "initiateShipping",
        "flow30": "parallelgateway1",
        "flow12": "card",
        "flow56": "parallelgateway5",
        "flow35": "initiateDelivery",
        "flow17": "exclusivegateway3",
        "flow63": "endevent1",
        "flow41": "notifyUser",
        "flow58": "sms",
        "flow5": "exclusivegateway2",
        "flow8": "exclusivegateway1",
        "flow33": "parallelgateway2",
        "flow11": "giftVoucher",
        "flow16": "validatePayment",
        "flow34": "parallelgateway2",
        "flow39": "parallelgateway3",
        "flow20": "errorendevent1",
        "flow32": "generateTrackingInfo",
        "flow10": "inclusivegateway1",
        "flow9": "Payment",
        "flow15": "inclusivegateway2",
        "flow3": "signIn",
        "flow47": "parallelgateway4",
        "flow21": "exclusivegateway4"
    },
    "name": "e017OnlineOrdering",
    "attached": {},
    "containE": {
        "Process_": [
            "flow63",
            "flow61",
            "flow47",
            "flow60",
            "flow58",
            "flow56",
            "flow43",
            "flow41",
            "flow59",
            "flow57",
            "flow34",
            "flow33",
            "flow32",
            "flow31",
            "flow62",
            "flow39",
            "flow35",
            "flow30",
            "flow21",
            "flow20",
            "flow19",
            "flow17",
            "flow16",
            "flow15",
            "flow14",
            "flow13",
            "flow12",
            "flow11",
            "flow10",
            "flow9",
            "flow8",
            "flow7",
            "flow6",
            "flow5",
            "flow4",
            "flow3"
        ]
    },
    "edges": [
        "flow63",
        "flow61",
        "flow47",
        "flow60",
        "flow58",
        "flow56",
        "flow43",
        "flow41",
        "flow59",
        "flow57",
        "flow34",
        "flow33",
        "flow32",
        "flow31",
        "flow62",
        "flow39",
        "flow35",
        "flow30",
        "flow21",
        "flow20",
        "flow19",
        "flow17",
        "flow16",
        "flow15",
        "flow14",
        "flow13",
        "flow12",
        "flow11",
        "flow10",
        "flow9",
        "flow8",
        "flow7",
        "flow6",
        "flow5",
        "flow4",
        "flow3"
    ],
    "messages": [],
    "sourceE": {
        "flow6": "signIn",
        "flow60": "sms",
        "flow14": "giftVoucher",
        "flow19": "exclusivegateway3",
        "flow61": "parallelgateway6",
        "flow43": "parallelgateway3",
        "flow7": "startevent1",
        "flow57": "parallelgateway5",
        "flow31": "parallelgateway1",
        "flow13": "inclusivegateway1",
        "flow59": "email",
        "flow4": "exclusivegateway1",
        "flow62": "exclusivegateway4",
        "flow30": "initiateShipping",
        "flow12": "inclusivegateway1",
        "flow56": "notifyUser",
        "flow35": "parallelgateway2",
        "flow17": "validatePayment",
        "flow63": "parallelgateway4",
        "flow41": "parallelgateway3",
        "flow58": "parallelgateway5",
        "flow5": "signUp",
        "flow8": "selectItems",
        "flow33": "generateTrackingInfo",
        "flow11": "inclusivegateway1",
        "flow16": "inclusivegateway2",
        "flow34": "packageItems",
        "flow39": "initiateDelivery",
        "flow20": "exclusivegateway3",
        "flow32": "parallelgateway1",
        "flow10": "Payment",
        "flow9": "exclusivegateway2",
        "flow15": "card",
        "flow3": "exclusivegateway1",
        "flow47": "transport",
        "flow21": "payOnDelivery"
    },
    "nodes": [
        "Process_",
        "exclusivegateway4",
        "exclusivegateway3",
        "validatePayment",
        "inclusivegateway2",
        "payOnDelivery",
        "card",
        "giftVoucher",
        "inclusivegateway1",
        "Payment",
        "selectItems",
        "exclusivegateway2",
        "signUp",
        "signIn",
        "exclusivegateway1",
        "startevent1",
        "initiateShipping",
        "parallelgateway1",
        "packageItems",
        "parallelgateway2",
        "initiateDelivery",
        "email",
        "notifyUser",
        "sms",
        "transport",
        "parallelgateway5",
        "parallelgateway6",
        "parallelgateway3",
        "parallelgateway4",
        "endevent1",
        "generateTrackingInfo",
        "errorendevent1"
    ],
    "nameN": {
        "giftVoucher": "Gift Voucher",
        "card": "Card",
        "validatePayment": "Validate Payment",
        "email": "Email",
        "packageItems": "Package Items",
        "signIn": "Sign In",
        "initiateDelivery": "Initiate Delivery",
        "generateTrackingInfo": "Generate Tracking Info",
        "Payment": "Payment",
        "startevent1": "Start",
        "signUp": "Sign Up",
        "notifyUser": "Notify User",
        "sms": "SMS",
        "payOnDelivery": "Pay On delivery",
        "transport": "Transport",
        "selectItems": "Select Items",
        "initiateShipping": "Initiate Shipping"
    }
}