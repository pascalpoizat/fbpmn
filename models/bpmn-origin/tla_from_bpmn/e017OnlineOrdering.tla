---------------- MODULE e017OnlineOrdering ----------------

EXTENDS TLC, PWSTypes

VARIABLES nodemarks, edgemarks, net

Interest ==
  "Process_" :> {  }

ContainRel ==
  "Process_" :> { "exclusivegateway4", "exclusivegateway3", "validatePayment", "inclusivegateway2", "payOnDelivery", "card", "giftVoucher", "inclusivegateway1", "Payment", "selectItems", "exclusivegateway2", "signUp", "signIn", "exclusivegateway1", "startevent1", "initiateShipping", "parallelgateway1", "packageItems", "parallelgateway2", "initiateDelivery", "email", "notifyUser", "sms", "transport", "parallelgateway5", "parallelgateway6", "parallelgateway3", "parallelgateway4", "endevent1", "generateTrackingInfo", "errorendevent1" }

Node == { "Process_", "exclusivegateway4", "exclusivegateway3", "validatePayment", "inclusivegateway2", "payOnDelivery", "card", "giftVoucher", "inclusivegateway1", "Payment", "selectItems", "exclusivegateway2", "signUp", "signIn", "exclusivegateway1", "startevent1", "initiateShipping", "parallelgateway1", "packageItems", "parallelgateway2", "initiateDelivery", "email", "notifyUser", "sms", "transport", "parallelgateway5", "parallelgateway6", "parallelgateway3", "parallelgateway4", "endevent1", "generateTrackingInfo", "errorendevent1" }

Edge == { "flow63", "flow61", "flow47", "flow60", "flow58", "flow56", "flow43", "flow41", "flow59", "flow57", "flow34", "flow33", "flow32", "flow31", "flow62", "flow39", "flow35", "flow30", "flow21", "flow20", "flow19", "flow17", "flow16", "flow15", "flow14", "flow13", "flow12", "flow11", "flow10", "flow9", "flow8", "flow7", "flow6", "flow5", "flow4", "flow3" }

Message == {  }

msgtype ==
  [ i \in {} |-> {}]


source ==
   "flow63" :> "parallelgateway4"
@@ "flow61" :> "parallelgateway6"
@@ "flow47" :> "transport"
@@ "flow60" :> "sms"
@@ "flow58" :> "parallelgateway5"
@@ "flow56" :> "notifyUser"
@@ "flow43" :> "parallelgateway3"
@@ "flow41" :> "parallelgateway3"
@@ "flow59" :> "email"
@@ "flow57" :> "parallelgateway5"
@@ "flow34" :> "packageItems"
@@ "flow33" :> "generateTrackingInfo"
@@ "flow32" :> "parallelgateway1"
@@ "flow31" :> "parallelgateway1"
@@ "flow62" :> "exclusivegateway4"
@@ "flow39" :> "initiateDelivery"
@@ "flow35" :> "parallelgateway2"
@@ "flow30" :> "initiateShipping"
@@ "flow21" :> "payOnDelivery"
@@ "flow20" :> "exclusivegateway3"
@@ "flow19" :> "exclusivegateway3"
@@ "flow17" :> "validatePayment"
@@ "flow16" :> "inclusivegateway2"
@@ "flow15" :> "card"
@@ "flow14" :> "giftVoucher"
@@ "flow13" :> "inclusivegateway1"
@@ "flow12" :> "inclusivegateway1"
@@ "flow11" :> "inclusivegateway1"
@@ "flow10" :> "Payment"
@@ "flow9" :> "exclusivegateway2"
@@ "flow8" :> "selectItems"
@@ "flow7" :> "startevent1"
@@ "flow6" :> "signIn"
@@ "flow5" :> "signUp"
@@ "flow4" :> "exclusivegateway1"
@@ "flow3" :> "exclusivegateway1"

target ==
   "flow63" :> "endevent1"
@@ "flow61" :> "parallelgateway4"
@@ "flow47" :> "parallelgateway4"
@@ "flow60" :> "parallelgateway6"
@@ "flow58" :> "sms"
@@ "flow56" :> "parallelgateway5"
@@ "flow43" :> "transport"
@@ "flow41" :> "notifyUser"
@@ "flow59" :> "parallelgateway6"
@@ "flow57" :> "email"
@@ "flow34" :> "parallelgateway2"
@@ "flow33" :> "parallelgateway2"
@@ "flow32" :> "generateTrackingInfo"
@@ "flow31" :> "packageItems"
@@ "flow62" :> "initiateShipping"
@@ "flow39" :> "parallelgateway3"
@@ "flow35" :> "initiateDelivery"
@@ "flow30" :> "parallelgateway1"
@@ "flow21" :> "exclusivegateway4"
@@ "flow20" :> "errorendevent1"
@@ "flow19" :> "exclusivegateway4"
@@ "flow17" :> "exclusivegateway3"
@@ "flow16" :> "validatePayment"
@@ "flow15" :> "inclusivegateway2"
@@ "flow14" :> "inclusivegateway2"
@@ "flow13" :> "payOnDelivery"
@@ "flow12" :> "card"
@@ "flow11" :> "giftVoucher"
@@ "flow10" :> "inclusivegateway1"
@@ "flow9" :> "Payment"
@@ "flow8" :> "exclusivegateway1"
@@ "flow7" :> "selectItems"
@@ "flow6" :> "exclusivegateway2"
@@ "flow5" :> "exclusivegateway2"
@@ "flow4" :> "signUp"
@@ "flow3" :> "signIn"

CatN ==
   "Process_" :> Process
@@ "exclusivegateway4" :> ExclusiveOr
@@ "exclusivegateway3" :> ExclusiveOr
@@ "validatePayment" :> AbstractTask
@@ "inclusivegateway2" :> InclusiveOr
@@ "payOnDelivery" :> AbstractTask
@@ "card" :> AbstractTask
@@ "giftVoucher" :> AbstractTask
@@ "inclusivegateway1" :> InclusiveOr
@@ "Payment" :> AbstractTask
@@ "selectItems" :> AbstractTask
@@ "exclusivegateway2" :> ExclusiveOr
@@ "signUp" :> AbstractTask
@@ "signIn" :> AbstractTask
@@ "exclusivegateway1" :> ExclusiveOr
@@ "startevent1" :> NoneStartEvent
@@ "initiateShipping" :> AbstractTask
@@ "parallelgateway1" :> Parallel
@@ "packageItems" :> AbstractTask
@@ "parallelgateway2" :> Parallel
@@ "initiateDelivery" :> AbstractTask
@@ "email" :> AbstractTask
@@ "notifyUser" :> AbstractTask
@@ "sms" :> AbstractTask
@@ "transport" :> AbstractTask
@@ "parallelgateway5" :> Parallel
@@ "parallelgateway6" :> Parallel
@@ "parallelgateway3" :> Parallel
@@ "parallelgateway4" :> Parallel
@@ "endevent1" :> NoneEndEvent
@@ "generateTrackingInfo" :> AbstractTask
@@ "errorendevent1" :> NoneEndEvent

CatE ==
   "flow63" :> NormalSeqFlow
@@ "flow61" :> NormalSeqFlow
@@ "flow47" :> NormalSeqFlow
@@ "flow60" :> NormalSeqFlow
@@ "flow58" :> NormalSeqFlow
@@ "flow56" :> NormalSeqFlow
@@ "flow43" :> NormalSeqFlow
@@ "flow41" :> NormalSeqFlow
@@ "flow59" :> NormalSeqFlow
@@ "flow57" :> NormalSeqFlow
@@ "flow34" :> NormalSeqFlow
@@ "flow33" :> NormalSeqFlow
@@ "flow32" :> NormalSeqFlow
@@ "flow31" :> NormalSeqFlow
@@ "flow62" :> NormalSeqFlow
@@ "flow39" :> NormalSeqFlow
@@ "flow35" :> NormalSeqFlow
@@ "flow30" :> NormalSeqFlow
@@ "flow21" :> NormalSeqFlow
@@ "flow20" :> ConditionalSeqFlow
@@ "flow19" :> DefaultSeqFlow
@@ "flow17" :> NormalSeqFlow
@@ "flow16" :> NormalSeqFlow
@@ "flow15" :> NormalSeqFlow
@@ "flow14" :> NormalSeqFlow
@@ "flow13" :> DefaultSeqFlow
@@ "flow12" :> ConditionalSeqFlow
@@ "flow11" :> ConditionalSeqFlow
@@ "flow10" :> NormalSeqFlow
@@ "flow9" :> NormalSeqFlow
@@ "flow8" :> NormalSeqFlow
@@ "flow7" :> NormalSeqFlow
@@ "flow6" :> NormalSeqFlow
@@ "flow5" :> NormalSeqFlow
@@ "flow4" :> DefaultSeqFlow
@@ "flow3" :> ConditionalSeqFlow

LOCAL preEdges ==
   <<"inclusivegateway2", "flow15">> :> {"flow10", "flow12", "flow3", "flow4", "flow5", "flow6", "flow7", "flow8", "flow9"}
@@ <<"inclusivegateway2", "flow14">> :> {"flow10", "flow11", "flow3", "flow4", "flow5", "flow6", "flow7", "flow8", "flow9"}
@@ <<"inclusivegateway1", "flow10">> :> {"flow3", "flow4", "flow5", "flow6", "flow7", "flow8", "flow9"}
PreEdges(n,e) == preEdges[n,e]

PreNodes(n,e) == { target[ee] : ee \in preEdges[n,e] }
          \union { nn \in { source[ee] : ee \in preEdges[n,e] } : CatN[nn] \in { NoneStartEvent, MessageStartEvent } }

BoundaryEvent ==
  [ i \in {} |-> {}]

WF == INSTANCE PWSWellFormed
ASSUME WF!WellTyped
ASSUME WF!WellFormedness

ConstraintNode == TRUE \* none
ConstraintEdge == TRUE \* none
Constraint == TRUE     \* none
INSTANCE PWSConstraints
INSTANCE UserProperties
INSTANCE PWSSemantics

================================================================

