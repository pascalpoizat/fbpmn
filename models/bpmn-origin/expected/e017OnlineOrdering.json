{
  "e017OnlineOrdering": {
    "Network02FifoPair": {
      "Prop02Safe": {
        "value": true,
        "states": "93",
        "trans": "141",
        "depth": "37"
      },
      "Prop04MsgSound": {
        "value": true,
        "states": "93",
        "trans": "141",
        "depth": "37"
      },
      "Prop03Sound": {
        "value": true,
        "states": "93",
        "trans": "141",
        "depth": "37"
      },
      "Prop01Type": {
        "value": true,
        "states": "93",
        "trans": "141",
        "depth": "37"
      }
    },
    "Network03Causal": {
      "Prop02Safe": {
        "value": true,
        "states": "93",
        "trans": "141",
        "depth": "37"
      },
      "Prop04MsgSound": {
        "value": true,
        "states": "93",
        "trans": "141",
        "depth": "37"
      },
      "Prop03Sound": {
        "value": true,
        "states": "93",
        "trans": "141",
        "depth": "37"
      },
      "Prop01Type": {
        "value": true,
        "states": "93",
        "trans": "141",
        "depth": "37"
      }
    },
    "Network07RSC": {
      "Prop02Safe": {
        "value": true,
        "states": "93",
        "trans": "141",
        "depth": "37"
      },
      "Prop04MsgSound": {
        "value": true,
        "states": "93",
        "trans": "141",
        "depth": "37"
      },
      "Prop03Sound": {
        "value": true,
        "states": "93",
        "trans": "141",
        "depth": "37"
      },
      "Prop01Type": {
        "value": true,
        "states": "93",
        "trans": "141",
        "depth": "37"
      }
    },
    "Network05Outbox": {
      "Prop02Safe": {
        "value": true,
        "states": "93",
        "trans": "141",
        "depth": "37"
      },
      "Prop04MsgSound": {
        "value": true,
        "states": "93",
        "trans": "141",
        "depth": "37"
      },
      "Prop03Sound": {
        "value": true,
        "states": "93",
        "trans": "141",
        "depth": "37"
      },
      "Prop01Type": {
        "value": true,
        "states": "93",
        "trans": "141",
        "depth": "37"
      }
    },
    "Network04Inbox": {
      "Prop02Safe": {
        "value": true,
        "states": "93",
        "trans": "141",
        "depth": "37"
      },
      "Prop04MsgSound": {
        "value": true,
        "states": "93",
        "trans": "141",
        "depth": "37"
      },
      "Prop03Sound": {
        "value": true,
        "states": "93",
        "trans": "141",
        "depth": "37"
      },
      "Prop01Type": {
        "value": true,
        "states": "93",
        "trans": "141",
        "depth": "37"
      }
    },
    "Network01Bag": {
      "Prop02Safe": {
        "value": true,
        "states": "93",
        "trans": "141",
        "depth": "37"
      },
      "Prop04MsgSound": {
        "value": true,
        "states": "93",
        "trans": "141",
        "depth": "37"
      },
      "Prop03Sound": {
        "value": true,
        "states": "93",
        "trans": "141",
        "depth": "37"
      },
      "Prop01Type": {
        "value": true,
        "states": "93",
        "trans": "141",
        "depth": "37"
      }
    },
    "Network06Fifo": {
      "Prop02Safe": {
        "value": true,
        "states": "93",
        "trans": "141",
        "depth": "37"
      },
      "Prop04MsgSound": {
        "value": true,
        "states": "93",
        "trans": "141",
        "depth": "37"
      },
      "Prop03Sound": {
        "value": true,
        "states": "93",
        "trans": "141",
        "depth": "37"
      },
      "Prop01Type": {
        "value": true,
        "states": "93",
        "trans": "141",
        "depth": "37"
      }
    }
  }
}
